plugins/processors/lotus_peer_id/peerid.go

```
@@ -0,0 +1,163 @@
+	}
+	if len(l.Config_APIListenAddr) == 0 {
+		l.Config_APIListenAddr = DefaultConfig_APIListenAddr
+	}
+	l.apiBackOffThrottle = time.Second * 10
+	l.curAPIBackOff = time.Unix(0,0)
+}
+
+func (l *LotusPeerIDTagger) setPeerID() error {
+	nodeAPI, nodeCloser, err := l.getAPIUsingLotusConfig()
+	if err != nil {
+		return err
+	}
+	defer nodeCloser()
+	pid, err := nodeAPI.ID(context.Background())
+	if err != nil {
+		return err
+	}
+	l.peerID = pid
+	return nil
+}
+
+func (l *LotusPeerIDTagger) setExcludeMap() {
+	ex := make(map[string]struct{})
+	for _, e := range l.Excludes {
+		ex[e.Measurement] = struct{}{}
+	}
+	l.excludeMap = ex
+}
+
+func (l *LotusPeerIDTagger) getAPIUsingLotusConfig() (api.FullNode, func(), error) {
+	var (
+		nodeAPI    api.FullNode
+		nodeCloser func()
+	)
+	if len(l.Config_APIListenAddr) > 0 && len(l.Config_APIToken) > 0 {
+		api, apiCloser, err := rpc.GetFullNodeAPIUsingCredentials(l.Config_APIListenAddr, l.Config_APIToken)
+		if err != nil {
+			err = fmt.Errorf("connect with credentials: %v", err)
+			return nil, nil, err
+		}
+		nodeAPI = api
+		nodeCloser = apiCloser
+	} else {
+		api, apiCloser, err := rpc.GetFullNodeAPI(l.Config_DataPath)
+		if err != nil {
+			err = fmt.Errorf("connect from lotus state: %v", err)
+			return nil, nil, err
+		}
+		nodeAPI = api
+		nodeCloser = apiCloser
+	}
+	return nodeAPI, nodeCloser, nil
+}
+
+func (l *LotusPeerIDTagger) SampleConfig() string {
+	return sampleConfig
+}
+
+func (l *LotusPeerIDTagger) Description() string {
+	return "Decorate measurements that pass through this filter with the peer id of lotus daemon."
+}
+
+func (l *LotusPeerIDTagger) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	if !l.initializedDefaults {
+		// configure lotus connection
+		l.setAPIConfigDefaults()
+		// create mapping of excluded metrics for faster processing.
+		l.setExcludeMap()
+		l.initializedDefaults = true
+	}
+
+	if time.Since(l.curAPIBackOff) < l.apiBackOffThrottle{
+		return nil
```

I expected this requirement to be handled here: `If the Processor doesn't have the PeerID and is waiting to retry, the processor should raise an error once in the log, indicate the number of metrics which were not updated, and bypass the loop for that call to Apply.`

Make sure that we return `in` instead of `nil`.